src/sentry/api/endpoints/organization_spans_aggregation.py

```diff
@@ -0,0 +1,230 @@
+from sentry.models import Organization
+from sentry.search.events.builder import QueryBuilder
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+EventSpan = namedtuple(
+    "EventSpan",
+    [
+        "span_id",
+        "is_segment",
+        "parent_span_id",
+        "group",
+        "group_raw",
+        "description",
+        "start_ms",
+        "duration",
+        "exclusive_time",
+    ],
+)
+
+
+class EventSpans(TypedDict):
+    transaction_id: str
+    spans: List[EventSpan]
+
+
+NULL_GROUP = "00"
+
+
+@region_silo_endpoint
+class OrganizationSpansAggregationEndpoint(OrganizationEventsEndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    aggregated_tree: Dict[str, Dict[str, Union[str, int, float]]] = {}
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:starfish-view", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        transaction = request.query_params.get("transaction", None)
+        if transaction is None:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST, data={"details": "Transaction not provided"}
+            )
+
+        builder = QueryBuilder(
+            dataset=Dataset.SpansIndexed,
+            params=params,
+            selected_columns=["transaction_id", "count()"],
+            query=f"transaction:{transaction}",
+            limit=100,
+        )
+        builder.columns.append(
+            Function(
+                "groupArray",
+                parameters=[
+                    Function(
+                        "tuple",
+                        parameters=[
+                            Column("span_id"),
+                            Column("is_segment"),
+                            Column("parent_span_id"),
+                            Column("group"),
+                            Column("group_raw"),
+                            Column("description"),
+                            Column("start_ms"),
+                            Column("duration"),
+                            Column("exclusive_time"),
+                        ],
+                    )
+                ],
+                alias="spans",
+            )
+        )
+        snql_query = builder.get_snql_query()
+        snql_query.tenant_ids = {"organization_id": organization.id}
+        results = raw_snql_query(snql_query, Referrer.API_ORGANIZATION_SPANS_AGGREGATION.value)
+        for event in results["data"]:
+            span_tree = {}
+            root_span_id = None
+            spans = event["spans"]
+
+            for span in spans:
+                span = EventSpan(*span)
```

Comment:
Could we rename the variable here instead of overwriting what's being iterated on?

Reply:
https://github.com/getsentry/sentry/pull/56291/commits/1fbbfe05516aba2dd890cb846b44a7af04a3de77#diff-1df1994c8fddccf95629e8b491f1ff17c43534b3634dd396256fc448124d7cddR105